interface/billing/billing_report.php

```diff
@@ -1142,14 +1144,15 @@ function criteriaSelectHasValue(select) {
                                     $last_level_closed = sqlQuery("SELECT `last_level_closed` FROM `form_encounter` WHERE `encounter` = ?", array($iter['enc_encounter']))['last_level_closed'];
                                     $effective_insurances = getEffectiveInsurances($iter['pid'], $iter['enc_date']);
                                     $insuranceCount = count($effective_insurances ?? []);
-
+                                    $default_x12_partner = null;
                                     foreach ($effective_insurances as $key => $row) {
                                         $insuranceName = sqlQuery("SELECT `name` FROM `insurance_companies` WHERE `id` = ?", array($row['provider']))['name'];
                                         $x12Partner = sqlQuery("SELECT `x12_default_partner_id` FROM `insurance_companies` WHERE `id` = ?", array($row['provider']))['x12_default_partner_id'];
                                         $lhtml .= "<option value=\"" . attr(substr($row['type'], 0, 1) . $row['provider']) . "\"";
                                         if (
-                                            $key == $last_level_closed
-                                            || $insuranceCount == 1
+                                            is_array($effective_insurances) &&
+                                            !empty($effective_insurances) &&
+                                            ($key == $last_level_closed || $insuranceCount == 1)
```

Comment:
ok, nice, thanks. no need then to check on `$effective_insurances` inside the loop

Reply:
it's minor bu this should be fixed